web3/middleware/stalecheck.py

```diff
@@ -0,0 +1,44 @@
+import time
+
+from web3.exceptions import StaleBlockchain
+
+SKIP_STALECHECK_FOR_METHODS = set([
+    'eth_getBlockByNumber',
+])
+
+
+def _isfresh(block, allowable_delay):
+    return block and time.time() - block['timestamp'] <= allowable_delay
+
+
+def make_stalecheck_middleware(allowable_delay=0):
+    '''
+    Use to require that a function will run only of the blockchain is recently updated.
+
+    This middleware takes an argument, so unlike other middleware, you must make the middleware
+    with a method call.
+    For example: `make_stalecheck_middleware(allowable_delay=60*5)`
+
+    If the latest block in the chain is older than 5 minutes in this example, then the
+    middleware will raise a StaleBlockchain exception.
+    '''
+    if not allowable_delay:
+        raise ValueError("You must set a non-zero allowable delay for this middleware")
+
+    def stalecheck_middleware(make_request, web3):
+        cache = {'latest': None}
+
+        def middleware(method, params):
+            if method not in SKIP_STALECHECK_FOR_METHODS:
```

Comment:
Can you add a new keyword argument to the factory, something like `skip_stalecheck_for_methods=SKIP_STALECHECK_FOR_METHODS` so that users can pass in their own list for this?

Reply:
Yeah, you mentioned this on the first review. I still don't understand when someone would use it. I'll add for it for now, but I'm curious what case you have in mind.

Reply:
It seems like someone might want to either add more methods, or remove methods from that list.  Given the uncertainty about which methods are most appropriate to skip or not skip this seems like a good feature to have (even if you don't document it) to allow work arounds for special cases were they to arise...